packages/plugin-browser-device/device.js

```diff
@@ -19,6 +43,15 @@ module.exports = (nav = navigator, screen = window.screen) => ({
           : 'portrait'
     }
 
+    if (client._config.generateAnonymousId) {
+      const id = getDeviceId()
+
+      // If there's no ID, it's likely because localStorage is disabled
+      if (id) {
+        device.id = id
```

Comment:
as long as `getDeviceId()` returns an `id` or `undefined`, I think it's safe enough and terser to do:

`device.id = getDeviceId()`